### PR TITLE
Add remaining user-methods for bindgen

### PR DESCRIPTION
**Pull request by JacobOscarGunnarsson**
Standalone PR which complements #4808 and #4805 now that support for Bson is provided through the bindgen.

implements the following methods:

- `User.customData()`
- `User.refreshCustomData()`
- `User.profile()`
- `EmailPasswordAuthClient.callResetPassWordFunction()`

### CODE REVIEW

File: packages/realm/src/app-services/User.ts [hunk 1]
```diff
@@ -70,6 +70,9 @@ export class User<
   /** @internal */
   public internal: binding.SyncUser;
 
+  // cached version of profile
+  private _profile: UserProfileDataType | undefined;
```
Comment (JacobOscarGunnarsson) [inline packages/realm/src/app-services/User.ts:74]: we talked about not using underscore for private fields but I'm not sure how we actually want it so putting it out there like this for now

File: packages/realm/src/app-services/User.ts [hunk 2]
```diff
@@ -163,14 +167,21 @@ export class User<
    * If this value has not been configured, it will be empty.
    */
   get customData(): CustomDataType {
-    throw new Error("Not yet implemented");
+    const result = this.internal.customData;
```
Comment (JacobOscarGunnarsson) [inline packages/realm/src/app-services/User.ts:170]: maybe it would be worth having a cached type for this as well?

Comment (kraenhansen) [reply packages/realm/src/app-services/User.ts:170]: It gets updated when the access token gets refreshed, so I think it's better to leave it un-memoized for now.
Also, it's not too expensive compared to how little it'll get called and end-users can always work around by caching it on the calling side if they experience performance issues.

File: packages/realm/src/app-services/User.ts (hunk 1 above)
Comment (takameyer) [reply packages/realm/src/app-services/User.ts:74]: I would name it `cachedProfile` and remove the `_`.

File: packages/realm/src/app-services/User.ts [hunk 3]
```diff
@@ -163,14 +167,21 @@ export class User<
...
+    }
+    return result as CustomDataType;
   }
 
   /**
    * A profile containing additional information about the user.
    */
   get profile(): UserProfileDataType {
-    throw new Error("Not yet implemented");
+    if (!this._profile) {
```
Comment (takameyer) [inline packages/realm/src/app-services/User.ts:181]: Does this cached `_profile` get invalidated somehow?

**Review by takameyer — APPROVED** [review]
Just a few comments, but nothing holding this back.

**Pull request merged**